src/Lib/V3/Tx.ts

```diff
@@ -0,0 +1,154 @@
+/**
+ * @see {@link https://github.com/input-output-hk/plutus/blob/1.36.0.0/plutus-ledger-api/src/PlutusLedgerApi/V1/Tx.hs}
+ */
+
+import type { Eq, Integer, Json, Maybe } from "prelude";
+import type { IsPlutusData } from "../PlutusData.js";
+import { IsPlutusDataError } from "../PlutusData.js";
+import * as PreludeInstances from "../Prelude/Instances.js";
+import * as Prelude from "prelude";
+import { JsonError } from "prelude";
+import * as LbBytes from "../V1/Bytes.js";
+
+/**
+ * {@link TxId} is a transaction id i.e., the hash of a transaction. 32 bytes.
+ *
+ * @see {@link https://github.com/input-output-hk/plutus/blob/1.36.0.0/plutus-ledger-api/src/PlutusLedgerApi/V3/Tx.hs#L42-L54}
+ */
+export type TxId = LbBytes.LedgerBytes & { __compileTimeOnlyTxId: TxId };
+
+/**
+ * {@link txIdFromBytes} checks if the bytes are 32 bytes long.
+ */
+export function txIdFromBytes(bytes: LbBytes.LedgerBytes): Maybe<TxId> {
+  if (bytes.length === 32) {
+    return { fields: bytes as TxId, name: "Just" };
+  } else {
+    return { name: "Nothing" };
+  }
+}
+
+/**
+ * {@link Eq} instance for {@link TxId}
+ */
+export const eqTxId: Eq<TxId> = LbBytes.eqLedgerBytes as Eq<TxId>;
+
+/**
+ * {@link Json} instance for {@link TxId}
+ */
+export const jsonTxId: Json<TxId> = {
+  toJson: LbBytes.jsonLedgerBytes.toJson,
+  fromJson: (value) => {
+    const bs = txIdFromBytes(LbBytes.jsonLedgerBytes.fromJson(value));
+    if (bs.name === "Nothing") {
+      throw new JsonError(`TxId should be 32 bytes`);
+    }
+    return bs.fields;
+  },
+};
+
+/**
+ * {@link IsPlutusData} instance for {@link TxId}
+ *
+ * Note this includes the `0` tag.
```

Comment:
```suggestion
```

Looks like the newer representation is a bit more efficient and doesn't include the tag. See the `deriving newtype `clause of https://github.com/input-output-hk/plutus/blob/1.36.0.0/plutus-ledger-api/src/PlutusLedgerApi/V3/Tx.hs#L42-L54}

Reply:
Ah, thank you, I forgot to update the docs